nextspeaker-web/src/main.rs

```diff
@@ -31,17 +42,54 @@ fn from_lines(text: &str) -> Result<Vec<String>> {
         .collect())
 }
 
+#[derive(Properties, PartialEq)]
+struct DismissableTextProps {
+    heading: String,
+    oninput: Callback<InputEvent>,
+    dismiss: Callback<MouseEvent>,
+    text: String,
+}
+
+#[styled_component]
+fn DismissableText(props: &DismissableTextProps) -> Html {
+    html! {
+        <div class={css!("background-color: lightgray; display: grid; width: 90%; padding: 1rem; grid-template-columns: 80% 20%;")}>
+            <Text heading={props.heading.clone()} text={props.text.clone()} oninput={props.oninput.clone()}></Text>
+            <button
+                class={css!("color: red; justify-self: right; align-self: start; height: 1.6rem;")}
+                onclick={props.dismiss.clone()}
+            >{"X"}</button>
+        </div>
+    }
+}
+
+#[derive(Properties, PartialEq)]
+struct ModeSelectProps {
+    candidates_view: Callback<MouseEvent>,
+    history_view: Callback<MouseEvent>,
+}
+
+#[styled_component]
+fn ModeSelect(props: &ModeSelectProps) -> Html {
```

Comment:
I think there ought to be a `ModeButton` component here?
We can keep `ModeSelect` too, as it is the `ButtonRow` we discussed earlier out of band.

Reply:
I pushed a change that makes the buttons a property passed to the `ModeSelect`, but because I can't yet see a difference between an HTML5 `button` and a `ModeButton`, I'm deferring the creation of another custom component.

Reply:
I am thinking of `ModeButton` as a logical component that knows what mode it should be toggling and whether it should be enabled (because of a prop from the parent, not because it inspects the global state directly). It would also need a cb for the actual mode setting behavior.

Reply:
With the latest commits, it doesn't have to know things, because they're supplied as properties.  I think if/when I want to style buttons differently in styled components, I'll stop using vanilla `button`s.